src/control.rs

```diff
@@ -1,25 +1,72 @@
+use crate::event_loop::EDT;
+use crate::event_loop::Msg;
 use crate::led::Led;
 use crate::pin::Pin;
+use crate::rgb::{BLUE, GREEN, RED, Rgb};
 
 /// Control logic evaluates button states and changes the light intensity
 pub struct LightControl<'a, P: Pin> {
     pub plus_pin: P,
     pub minus_pin: P,
     pub led: &'a dyn Led,
+    pub rgb: &'a dyn Rgb,
+    pub edt: &'a EDT,
 }
 
-impl <'a, P: Pin> LightControl<'a, P> {
-    pub fn tick(&self) {
+const MSG_BLINK: i32 = 1;
+const MSG_CHECK_BUTTONS: i32 = 2;
+
+const DELAY_CHECK_BUTTONS: u32 = 75;
+const DELAY_BLINK: u32 = 250;
+
+impl<'a, P: Pin> LightControl<'a, P> {
+    pub fn process_message(&self, msg: Msg) {
+        match msg.what {
+            MSG_CHECK_BUTTONS => self.check_buttons(),
+            MSG_BLINK => self.blink_led(msg),
+            _ => {}
+        }
+    }
+
+    fn blink_led(&self, msg: Msg) {
+        if msg.arg1 > 0 {
+            let rgb = self.rgb.get_rgb();
+            self.rgb.set_rgb(rgb ^ (msg.arg0 as u8));
+            self.edt.schedule_with_args(
+                DELAY_BLINK,
+                MSG_BLINK,
+                msg.arg0,
+                msg.arg1 - 1,
+            );
+        }
+    }
+
+    pub fn check_buttons(&self) {
         if self.plus_pin.is_down() {
             self.led.modify(&|current: u8| {
-                if current < 8 { current + 1 } else { current }
+                if current < 32 { current + 1 } else { current }

```

Comment:
32 - Magic number. Maybe use const there?

Reply:
This has to be replaced with a proper 2^n sequence until the maximum PWM value is reached. Something like:

1 2 4 8 16 32 64 128 255

Or maybe we will use 128 and 255 as "boost" values. Anyway, I need to know what is the PWM precision.

@siralexeyduke Is it 8 or 10 bit?

Reply:
Actually 16 bit, but I thought, that it should be abstracted away, so that you set the PWM duty-cycle in % and the underlying BSP does the conversion.

Reply:
For example:
```rust
static PWM_STEPS: &'static [i32] = &[0, 5, 15, 30, 60, 100];

...

if current < PWM_STEPS.len() { current + 1 } else { current }
```

Reply:
Yeah, and feel free to use u32 and i32, it is actually more efficient for ARM architecture as the u8, i8, u16 and i16.